gslib/tests/test_cp.py

```diff
@@ -1189,14 +1200,17 @@ def test_request_reason_header_persists_multiple_requests_json(self):
 
     boto_config_for_test = ('GSUtil', 'resumable_threshold', '0')
     with SetBotoConfigForTest([boto_config_for_test]):
-      stderr = self.RunGsUtil(['-D', 'cp', fpath, dst_uri], return_stderr=True)
+      stderr = self.RunGsUtil(['-DD', 'cp', fpath, dst_uri], return_stderr=True)
 
-    # PUT follows GET request. Both need the request-reason header.
-    reason_regex = (r'Making http GET[\s\S]*'
-                    r'x-goog-request-reason\': \'b/this_is_env_reason[\s\S]*'
-                    r'send: (b\')?PUT[\s\S]*x-goog-request-reason:'
-                    r' b/this_is_env_reason')
-    self.assertRegex(stderr, reason_regex)
+    if self._use_gcloud_storage:
+      reason_regex = r'X-Goog-Request-Reason\': b\'b/this_is_env_reason'
+    else:
+      reason_regex = r'x-goog-request-reason\': \'b/this_is_env_reason'
+
+    self.assertRegex(
+        stderr,
+        # PUT follows GET request. Both need the request-reason header.
+        r'GET[\s\S]*' + reason_regex + r'[\s\S]*PUT[\s\S]*' + reason_regex)
```

Comment:
Be careful that there are no other `GET` or `PUT` requests in `stderr` that might cause this to pass

Reply:
Definitely! There aren't any others as it stands. It's a bit of a risk that this might change, but I think really what this test has to cover is that multiple requests use the reason header.